### DISCUSSION TITLE
"free up memory" button?

### DISCUSSION THREAD

**Discussion by spxxkyy** [seerge/g-helper#2606] [Ideas]
There is a button in myasus that cleans up your memory, i use it pretty often since im stuck with 8 gigs of ram, but i find it fustrating how i have to open myasus and wait for it to load everytime i want to use it, i think it would be a good feature to add this button in ghelper for quicker access

**Comment by seerge** [comment]
@snifuh to my best knowledge all this "memory optimizers" doesn't make much sense, as modern windows manages memory quite well. And frees it up when it needs it (and not immediately after something is stopped).

But there are quite a few free "memory cleaner" apps that you can google and use :)

**Reply by spxxkyy** [reply]
Oh i see, alright thank you

great program btw, really useful

**Comment by muqadasah** [comment]
The "free up memory" button? does it clear temp files only? I am afraid of using it. will it clear any of my data in program files?

**Comment by Sibobby1** [comment]
I know this topic was already discussed, and I understand the usual argument that modern Windows memory management is generally good and that “memory optimizers” are often unnecessary.

My use case is a little more specific: I have a 2024 ASUS G14 with the RTX 4060 and 16 GB of soldered RAM, so the system is fairly RAM-constrained. Most of the time, Windows using available memory aggressively is fine and even beneficial.

The issue comes up when I need to start a Hyper-V virtual machine. I allocate 4 GB of RAM to the VM, and in practice I need enough available memory before Hyper-V will reliably launch it. Right now my workaround is:

1. Open MyASUS
2. Run “System Diagnosis / Repair”
3. Use the “Free Up Memory” option, sometimes more than once, until I have enough free RAM
4. Start the VM
5. Go back to G-Helper and stop ASUS tasks
6. Continue working

That workflow works, but it is clunky because I otherwise prefer to keep MyASUS/Armoury Crate-related services stopped and use G-Helper as the lightweight control utility.

I’m not asking for a general-purpose “RAM booster” or claiming Windows needs constant memory cleaning. I’m more asking whether G-Helper could expose a simple manual “free up memory” action, perhaps under **Extras**, for cases where a user intentionally wants to trigger memory cleanup before launching a VM, game, or other memory-heavy task.

Even if this just replicates the behavior of the MyASUS button, it would be useful because it avoids launching MyASUS just to perform that one action.

Totally understand if this is outside the scope of G-Helper, but I wanted to provide a concrete workflow where the feature would be useful beyond generic memory optimization.

**Comment by seerge** [comment]
@spxxkyy hello,

"Free up memory" is not a simple action that you can call remotely somehow, and let firmware / or other app do it for you. I don't know how exactly it's done in MyASUS - but has to force garbage collection for memory and  unload existing app recources into swap (i.e. to SSD)

If you have some exotic usecase and still use it - why not to keep doing it from MyASUS ? Or any other standalone "free up memory" apps ? :)

**Comment by Reddot911** [comment]
you can check out ISLC
